recipe/gen_patch_json.py

```diff
@@ -994,6 +994,17 @@ def _gen_new_index(repodata, subdir):
             new_depends = record.get("depends", [])
             new_depends.append("libgcc-ng <=9.3.0")
             record["depends"] = new_depends
+        
+        # setutools started raising a warning when using `LooseVersion` from distutils
+        # since packages don't tend to pin setuptools, this raises warnings in old versions
+        # https://github.com/conda-forge/conda-forge.github.io/issues/1575
+        if (
+            record_name in ["pandas", "distributed", "dask"]
```

Comment:
Looks like `dask` doesn't list this as a dependency. Only `distributed` does

```suggestion
            record_name == "distributed"
```

Reply:
The python dependency from conda-forge pulls in setuptools implicitly even if it's not listed. So I think we should actually be adding this dependency to dask-core.

```suggestion
            record_name in ["dask-core", "distributed"]
```

Reply:
`setuptools` is [not a `python` dependency]( https://github.com/conda-forge/python-feedstock/blob/1bb50920fe462e1a30ebda8c3fdabb30fe4bde0c/recipe/meta.yaml#L183-L185 ).

That said, [`conda` can pull in `pip` when installing `python`]( https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#add-pip-as-python-dependency-add-pip-as-python-dependency ) and [`pip` does pull in `setuptools`]( https://github.com/conda-forge/pip-feedstock/blob/774427c945373fa7ff1386019c57635301a35019/recipe/meta.yaml#L25 ). So guessing this is what you referring to.

Looking at Dask itself, it doesn't appear to use [`pkg_resources`]( https://github.com/dask/dask/search?q=pkg_resources ) or [`distutils`]( https://github.com/dask/dask/search?q=distutils ). The only [uses of `setuptools` are in `setup.py` and `versioneer`]( https://github.com/dask/dask/search?p=1&q=setuptools ). So it seems that if we were to add a constraint, we should add a `setuptools` constraint to `requirements/host` in `dask-core`.

Reply:
@jrbourbeau, do you have any thoughts here?

Reply:
Ok I changed this to do dask-core rather than dask.